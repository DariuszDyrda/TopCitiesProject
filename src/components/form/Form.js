import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button'
import Input from './Input';
import { strings } from '../../consts/strings';

const useStyles = makeStyles(() => ({
    root: {
        display: 'flex',
        flexGrow: 1,
        width: '100%',
        marginBottom: '3px',
        marginTop: '30px',
    },
}))

export const Form = (props) => {
    const [inputValue, setInputValue] = useState(null);
    const [buttonDisabled, setButtonDisabled] = useState(true);

    const classes = useStyles();

    return (
        <form data-testid={'form'} noValidate autoComplete="off" onSubmit={props.handleSubmit.bind(this, inputValue)} className={classes.root}>
            <Input setInputValue={setInputValue} setButtonDisabled={setButtonDisabled} />
            <Button disabled={buttonDisabled} variant="contained" type='submit' color="primary">{ strings.FORM_BUTTON_TEXT }</Button>
        </form>
    )
}

Form.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
}