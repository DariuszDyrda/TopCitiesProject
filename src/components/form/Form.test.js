import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import { Form } from "./Form";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("should render a form", () => {
  act(() => {
    render(<Form />, container);
  });
  expect(container).toMatchSnapshot();

});