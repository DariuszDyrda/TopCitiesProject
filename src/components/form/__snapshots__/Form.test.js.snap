// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a form 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-root-2"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-options-label"
        class="makeStyles-container-3"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="downshift-options-input"
            id="downshift-options-label"
          >
            Countries
          </label>
          <div
            class="MuiInputBase-root MuiInput-root makeStyles-inputRoot-5 MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              aria-labelledby="downshift-options-label"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input makeStyles-inputInput-6"
              id="downshift-options-input"
              placeholder="Country"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          aria-labelledby="downshift-options-label"
          id="downshift-options-menu"
          role="listbox"
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Search
      </span>
    </button>
  </div>
</div>
`;
